### ISSUE TITLE
Nvidia package does not compile with kernel 4.16.13 (Bugzilla Bug 2351)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#2351]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 2351
Date: 2018-06-08 02:28:40 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;
To:   @itchka
CC:   blackcrack@blackysgate.de, m-tomatis@tiscali.it, nix.or.die@gmail.com

Last updated: 2018-07-27 03:03:18 +0000

## Comment 18234
Date: 2018-06-08 02:28:40 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

## Comment 18235
Date: 2018-06-08 02:32:29 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Created attachment 1603
make for nvidia building in kernel 4.16.13

this is generated after 

# dkms install -m nvidia-current -v 390.25-1

> Attached file: nvidiamakelog.txt (text/plain, 207349 bytes)
> Description:   make for nvidia building in kernel 4.16.13

## Comment 18236
Date: 2018-06-08 02:34:41 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Created attachment 1604
make.log for building bbswitch in kernel 4.16.13

This is the output for 

# dkms install -m bbswitch -v 0.8-6

> Attached file: bbswitchmakelog.txt (text/plain, 1501 bytes)
> Description:   make.log for building bbswitch in kernel 4.16.13

## Comment 18237
Date: 2018-06-08 02:36:19 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

This is to report new kernel 4.16.13 does not build nvidia and bbswitch

Thanks

## Comment 18238
Date: 2018-06-08 02:45:53 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

Nvidia package is to old .. or one could patch the kernel.. with this:

https://raw.githubusercontent.com/frugalware/frugalware-current/master/source/base/kernel/0001-Revert-swiotlb-remove-various-exports.patch

## Comment 18239
Date: 2018-06-08 02:48:32 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

Also please one report per package..

That bbswitch compiles so what do you mean ?

## Comment 18240
Date: 2018-06-08 03:24:05 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Yes, one bug per package. I just thought the package with a bug is the kernel 4.16.13 itself.

Both nvidia and bbswitch does not build with kernel 4.16.13

## Comment 18241
Date: 2018-06-08 03:39:36 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to adelson from comment #6)
> Yes, one bug per package. I just thought the package with a bug is the
> kernel 4.16.13 itself.
> 
> Both nvidia and bbswitch does not build with kernel 4.16.13

Well from your log it builds ..

...

 MODPOST 1 modules
  CC      /var/lib/dkms/bbswitch/0.8-6/build/bbswitch.mod.o
  LD [M]  /var/lib/dkms/bbswitch/0.8-6/build/bbswitch.ko

...

the 'notes'you see are warnings not compile errors.

## Comment 18242
Date: 2018-06-08 04:00:38 +0000
From: @benbullard79

(In reply to adelson from comment #6)
> Yes, one bug per package. I just thought the package with a bug is the
> kernel 4.16.13 itself.
> 
> Both nvidia and bbswitch does not build with kernel 4.16.13

My  understanding is that the bug is not with the kernel package. The nVidia packages have to be made for a specific kernel version and this can only be done after the kernel is released. Further it is my understanding that some of that "making nvidia packages for a specific kernel" is dependent on nVidia the company making the code available. As they are not open source the Linux distros have no choice but to wait on nVidia the company to make the code available. That is the proprietary part of proprietary 3rd party software. It is their (nVidia's) code.

And I agree it is best to have separate bug reports. The nvidia drivers alone are more than enough for one bug report.

I believe I can confirm this bug just looking at the dates on OM's nvida packages in repos. Ain't no way they are gonna work with kernel 4.16.13.

## Comment 18243
Date: 2018-06-08 04:30:12 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Ben, notice that bbswitch is opensource and maintained by omv lx 3.03 and It does not build as well.

## Comment 18244
Date: 2018-06-08 05:19:36 +0000
From: @benbullard79

(In reply to adelson from comment #9)
> Ben, notice that bbswitch is opensource and maintained by omv lx 3.03 and It
> does not build as well.

If that is the case that is an example of why bbswitch should have it's own bug report and not be lumped in with nvidia drivers.

I haven't found a package bbswitch yet are you referring to dkms-bbswitch?

Either way I know next to nothing about bbswitch or dkms-bbswitch.

Also the fact that nvidia driver packages are in non-free repo does not mean "non-supported". Non-free repo is supported the distinction is supposed to be proprietary code or non-open source code hence non-free.

## Comment 18250
Date: 2018-06-08 12:17:41 +0000
From: @tpgxyz

According to your nvidia logs i found there no errors or anything that may claim that build failed, same goes to bbswitch logs.

Please provide logs that says there are some erros, anything that may help us to help you.

## Comment 18254
Date: 2018-06-08 13:01:26 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #11)
> According to your nvidia logs i found there no errors or anything that may
> claim that build failed, same goes to bbswitch logs.
> 
> Please provide logs that says there are some erros, anything that may help
> us to help you.

Hmm?

"swiotlb_map_sg_attrs" [/var/lib/dkms/nvidia-current/390.25-1/build/nvidia.ko] undefined!

broken won't ever load .. and is known issue , some kernel 
exported GPL symbols got removed.

Newer nvidia releases checks for that and have own workaround

## Comment 18256
Date: 2018-06-08 13:05:28 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

*** Bug #2329 has been marked as a duplicate of this bug. ***

## Comment 18259
Date: 2018-06-09 03:01:17 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #11)
> According to your nvidia logs i found there no errors or anything that may
> claim that build failed, same goes to bbswitch logs.
> 
> Please provide logs that says there are some erros, anything that may help
> us to help you.

About nvidia building, it seems that  Gabriel Craciunescu found some clue on the root of the problem...

The following is all that is output for bbswitch building and bbswitchmakelog.txt has all the content of :
/var/lib/dkms/bbswitch/0.8-6/build/make.log
where can I find more information?

" # dkms install -m bbswitch -v 0.8-6

  Kernel preparation unnecessary for this kernel.  Skipping...

  Building module:
  cleaning build area...
  make KERNELRELEASE=4.16.13-desktop-1omv KVERSION=4.16.13-desktop-1omv -j8...
  Error!  Build of bbswitch.ko.gz failed for: 4.16.13-desktop-1omv (x86_64)
  Consult the make.log in the build directory
  /var/lib/dkms/bbswitch/0.8-6/build/ for more information.  "

## Comment 18260
Date: 2018-06-09 03:36:28 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to adelson from comment #14)
> (In reply to Tomasz Paweł Gajc from comment #11)
> > According to your nvidia logs i found there no errors or anything that may
> > claim that build failed, same goes to bbswitch logs.
> > 
> > Please provide logs that says there are some erros, anything that may help
> > us to help you.
> 
> About nvidia building, it seems that  Gabriel Craciunescu found some clue on
> the root of the problem...
> 
> The following is all that is output for bbswitch building and
> bbswitchmakelog.txt has all the content of :
> /var/lib/dkms/bbswitch/0.8-6/build/make.log
> where can I find more information?
> 
> 
> " # dkms install -m bbswitch -v 0.8-6
> 
>   Kernel preparation unnecessary for this kernel.  Skipping...
> 
>   Building module:
>   cleaning build area...
>   make KERNELRELEASE=4.16.13-desktop-1omv KVERSION=4.16.13-desktop-1omv
> -j8...
>   Error!  Build of bbswitch.ko.gz failed for: 4.16.13-desktop-1omv (x86_64)
>   Consult the make.log in the build directory
>   /var/lib/dkms/bbswitch/0.8-6/build/ for more information.  "

I've no clue about bbswitch .. never used it nor do I use dkms for anything.

However , since you have everything in place already , can you please try
the following ?:

1) mkdir -p /tmp/bbtest && cd /tmp/bbtest
2) wget -c https://github.com/Bumblebee-Project/bbswitch/archive/v0.8.tar.gz 
3) tar -xvzf v0.8.tar.gz && cd bbswitch-0.8
4) make V=1 -j8 | tee bb.log

Thx :)

## Comment 18261
Date: 2018-06-09 04:03:09 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Created attachment 1605
bb.log

I guess the idea is to return the bb.log. It is in the attachment.

But, anyway, this has happened:

$ make V=1 -j8 | tee bb.log
make -C /lib/modules/4.16.13-desktop-1omv/build M="$(pwd)" modules
make[1]: Entering directory '/usr/src/linux-4.16.13-desktop-1omv'
test -e include/generated/autoconf.h -a -e include/config/auto.conf || (                \
echo >&2;                                                       \
echo >&2 "  ERROR: Kernel configuration is invalid.";           \
echo >&2 "         include/generated/autoconf.h or include/config/auto.conf are missing.";\
echo >&2 "         Run 'make oldconfig && make prepare' on kernel src to fix it.";      \
echo >&2 ;                                                      \
/bin/false)
mkdir -p /tmp/bbtest/bbswitch-0.8/.tmp_versions ; rm -f /tmp/bbtest/bbswitch-0.8/.tmp_versions/*
make -f ./scripts/Makefile.build obj=/tmp/bbtest/bbswitch-0.8
(cat /dev/null;   echo kernel//tmp/bbtest/bbswitch-0.8/bbswitch.ko;) > /tmp/bbtest/bbswitch-0.8/modules.order
  gcc -Wp,-MD,/tmp/bbtest/bbswitch-0.8/.bbswitch.o.d  -nostdinc -isystem /usr/lib64/gcc/x86_64-mandriva-linux-gnu/7.3.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -DCONFIG_CC_STACKPROTECTOR -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -fshort-wchar -Werror-implicit-function-declaration -Wno-format-security -std=gnu11 -fno-PIE -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64 -falign-jumps=1 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387 -mpreferred-stack-boundary=3 -mskip-rax-setup -mtune=generic -mno-red-zone -mcmodel=kernel -funit-at-a-time -DCONFIG_X86_X32_ABI -DCONFIG_AS_CFI=1 -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -DCONFIG_AS_CFI_SECTIONS=1 -DCONFIG_AS_FXSAVEQ=1 -DCONFIG_AS_SSSE3=1 -DCONFIG_AS_CRC32=1 -DCONFIG_AS_AVX=1 -DCONFIG_AS_AVX2=1 -DCONFIG_AS_AVX512=1 -DCONFIG_AS_SHA1_NI=1 -DCONFIG_AS_SHA256_NI=1 -pipe -Wno-sign-compare -fno-asynchronous-unwind-tables -mindirect-branch=thunk-extern -mindirect-branch-register -DRETPOLINE -fno-delete-null-pointer-checks -Wno-frame-address -Wno-format-truncation -Wno-format-overflow -Wno-int-in-bool-context -O2 --param=allow-store-data-races=0 -DCC_HAVE_ASM_GOTO -fplugin=./scripts/gcc-plugins/latent_entropy_plugin.so -fplugin=./scripts/gcc-plugins/structleak_plugin.so -fplugin=./scripts/gcc-plugins/randomize_layout_plugin.so -DLATENT_ENTROPY_PLUGIN -fplugin-arg-structleak_plugin-verbose -fplugin-arg-structleak_plugin-byref-all -DSTRUCTLEAK_PLUGIN -DRANDSTRUCT_PLUGIN -fplugin-arg-randomize_layout_plugin-performance-mode -Wframe-larger-than=2048 -fstack-protector -Wno-unused-but-set-variable -Wno-unused-const-variable -fno-var-tracking-assignments -pg -mfentry -DCC_USING_FENTRY -Wdeclaration-after-statement -Wno-pointer-sign -fno-strict-overflow -fno-merge-all-constants -fmerge-constants -fno-stack-check -fconserve-stack -Werror=implicit-int -Werror=strict-prototypes -Werror=date-time -Werror=incompatible-pointer-types -Werror=designated-init  -DMODULE  -DKBUILD_BASENAME='"bbswitch"'  -DKBUILD_MODNAME='"bbswitch"' -c -o /tmp/bbtest/bbswitch-0.8/.tmp_bbswitch.o /tmp/bbtest/bbswitch-0.8/bbswitch.c
/tmp/bbtest/bbswitch-0.8/bbswitch.c: In function ‘acpi_call_dsm’:
/tmp/bbtest/bbswitch-0.8/bbswitch.c:138:28: note: byref variable will be forcibly initialized
         struct acpi_buffer buf = { ACPI_ALLOCATE_BUFFER, NULL };
                            ^~~
/tmp/bbtest/bbswitch-0.8/bbswitch.c:109:29: note: byref variable will be forcibly initialized
     struct acpi_object_list input;
                             ^~~~~
/tmp/bbtest/bbswitch-0.8/bbswitch.c:108:24: note: byref variable will be forcibly initialized
     struct acpi_buffer output = { ACPI_ALLOCATE_BUFFER, NULL };
                        ^~~~~~
/tmp/bbtest/bbswitch-0.8/bbswitch.c: In function ‘bbswitch_init’:
/tmp/bbtest/bbswitch-0.8/bbswitch.c:398:28: note: byref variable will be forcibly initialized
         struct acpi_buffer buf = { ACPI_ALLOCATE_BUFFER, NULL };
                            ^~~
   ./tools/objtool/objtool orc generate  --module --no-fp  --retpoline "/tmp/bbtest/bbswitch-0.8/.tmp_bbswitch.o";
  if [ "-pg" = "-pg" ]; then if [ /tmp/bbtest/bbswitch-0.8/bbswitch.o != "scripts/mod/empty.o" ]; then ./scripts/recordmcount  "/tmp/bbtest/bbswitch-0.8/bbswitch.o"; fi; fi;
make -f ./scripts/Makefile.modpost
  find /tmp/bbtest/bbswitch-0.8/.tmp_versions -name '*.mod' | xargs -r grep -h '\.ko$' | sort -u | sed 's/\.ko$/.o/' | scripts/mod/modpost -m -a -i ./Module.symvers -I /tmp/bbtest/bbswitch-0.8/Module.symvers  -o /tmp/bbtest/bbswitch-0.8/Module.symvers -S  -w  -s -T -
  gcc -Wp,-MD,/tmp/bbtest/bbswitch-0.8/.bbswitch.mod.o.d  -nostdinc -isystem /usr/lib64/gcc/x86_64-mandriva-linux-gnu/7.3.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -DCONFIG_CC_STACKPROTECTOR -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -fshort-wchar -Werror-implicit-function-declaration -Wno-format-security -std=gnu11 -fno-PIE -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64 -falign-jumps=1 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387 -mpreferred-stack-boundary=3 -mskip-rax-setup -mtune=generic -mno-red-zone -mcmodel=kernel -funit-at-a-time -DCONFIG_X86_X32_ABI -DCONFIG_AS_CFI=1 -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -DCONFIG_AS_CFI_SECTIONS=1 -DCONFIG_AS_FXSAVEQ=1 -DCONFIG_AS_SSSE3=1 -DCONFIG_AS_CRC32=1 -DCONFIG_AS_AVX=1 -DCONFIG_AS_AVX2=1 -DCONFIG_AS_AVX512=1 -DCONFIG_AS_SHA1_NI=1 -DCONFIG_AS_SHA256_NI=1 -pipe -Wno-sign-compare -fno-asynchronous-unwind-tables -mindirect-branch=thunk-extern -mindirect-branch-register -DRETPOLINE -fno-delete-null-pointer-checks -Wno-frame-address -Wno-format-truncation -Wno-format-overflow -Wno-int-in-bool-context -O2 --param=allow-store-data-races=0 -DCC_HAVE_ASM_GOTO -fplugin=./scripts/gcc-plugins/latent_entropy_plugin.so -fplugin=./scripts/gcc-plugins/structleak_plugin.so -fplugin=./scripts/gcc-plugins/randomize_layout_plugin.so -DLATENT_ENTROPY_PLUGIN -fplugin-arg-structleak_plugin-verbose -fplugin-arg-structleak_plugin-byref-all -DSTRUCTLEAK_PLUGIN -DRANDSTRUCT_PLUGIN -fplugin-arg-randomize_layout_plugin-performance-mode -Wframe-larger-than=2048 -fstack-protector -Wno-unused-but-set-variable -Wno-unused-const-variable -fno-var-tracking-assignments -pg -mfentry -DCC_USING_FENTRY -Wdeclaration-after-statement -Wno-pointer-sign -fno-strict-overflow -fno-merge-all-constants -fmerge-constants -fno-stack-check -fconserve-stack -Werror=implicit-int -Werror=strict-prototypes -Werror=date-time -Werror=incompatible-pointer-types -Werror=designated-init  -DKBUILD_BASENAME='"bbswitch.mod"'  -DKBUILD_MODNAME='"bbswitch"' -DMODULE  -c -o /tmp/bbtest/bbswitch-0.8/bbswitch.mod.o /tmp/bbtest/bbswitch-0.8/bbswitch.mod.c
  ld -r -m elf_x86_64 -z max-page-size=0x200000 -T ./scripts/module-common.lds --build-id  -o /tmp/bbtest/bbswitch-0.8/bbswitch.ko /tmp/bbtest/bbswitch-0.8/bbswitch.o /tmp/bbtest/bbswitch-0.8/bbswitch.mod.o ;  true
ld: warning: Disabling --icf due to -r
make[1]: Leaving directory '/usr/src/linux-4.16.13-desktop-1omv'

> Attached file: bb.log (text/x-log, 6218 bytes)
> Description:   bb.log

## Comment 18262
Date: 2018-06-09 04:17:45 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to adelson from comment #16)
> Created attachment 1605 [details]
> bb.log
> 
> I guess the idea is to return the bb.log. It is in the attachment.
> 
> But, anyway, this has happened:
> 
> 
> $ make V=1 -j8 | tee bb.log
> make -C /lib/modules/4.16.13-desktop-1omv/build M="$(pwd)" modules
> make[1]: Entering directory '/usr/src/linux-4.16.13-desktop-1omv'
> test -e include/generated/autoconf.h -a -e include/config/auto.conf || (    
> \
> echo >&2;                                                       \
> echo >&2 "  ERROR: Kernel configuration is invalid.";           \
> echo >&2 "         include/generated/autoconf.h or include/config/auto.conf
> are missing.";\
> echo >&2 "         Run 'make oldconfig && make prepare' on kernel src to fix
> it.";      \
> echo >&2 ;                                                      \
> /bin/false)
> mkdir -p /tmp/bbtest/bbswitch-0.8/.tmp_versions ; rm -f
> /tmp/bbtest/bbswitch-0.8/.tmp_versions/*
> make -f ./scripts/Makefile.build obj=/tmp/bbtest/bbswitch-0.8
> (cat /dev/null;   echo kernel//tmp/bbtest/bbswitch-0.8/bbswitch.ko;) >
> /tmp/bbtest/bbswitch-0.8/modules.order
>   gcc -Wp,-MD,/tmp/bbtest/bbswitch-0.8/.bbswitch.o.d  -nostdinc -isystem
> /usr/lib64/gcc/x86_64-mandriva-linux-gnu/7.3.0/include -I./arch/x86/include
> -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi
> -I./arch/x86/include/generated/uapi -I./include/uapi
> -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__
> -DCONFIG_CC_STACKPROTECTOR -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs
> -fno-strict-aliasing -fno-common -fshort-wchar
> -Werror-implicit-function-declaration -Wno-format-security -std=gnu11
> -fno-PIE -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64
> -falign-jumps=1 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387
> -mpreferred-stack-boundary=3 -mskip-rax-setup -mtune=generic -mno-red-zone
> -mcmodel=kernel -funit-at-a-time -DCONFIG_X86_X32_ABI -DCONFIG_AS_CFI=1
> -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -DCONFIG_AS_CFI_SECTIONS=1
> -DCONFIG_AS_FXSAVEQ=1 -DCONFIG_AS_SSSE3=1 -DCONFIG_AS_CRC32=1
> -DCONFIG_AS_AVX=1 -DCONFIG_AS_AVX2=1 -DCONFIG_AS_AVX512=1
> -DCONFIG_AS_SHA1_NI=1 -DCONFIG_AS_SHA256_NI=1 -pipe -Wno-sign-compare
> -fno-asynchronous-unwind-tables -mindirect-branch=thunk-extern
> -mindirect-branch-register -DRETPOLINE -fno-delete-null-pointer-checks
> -Wno-frame-address -Wno-format-truncation -Wno-format-overflow
> -Wno-int-in-bool-context -O2 --param=allow-store-data-races=0
> -DCC_HAVE_ASM_GOTO -fplugin=./scripts/gcc-plugins/latent_entropy_plugin.so
> -fplugin=./scripts/gcc-plugins/structleak_plugin.so
> -fplugin=./scripts/gcc-plugins/randomize_layout_plugin.so
> -DLATENT_ENTROPY_PLUGIN -fplugin-arg-structleak_plugin-verbose
> -fplugin-arg-structleak_plugin-byref-all -DSTRUCTLEAK_PLUGIN
> -DRANDSTRUCT_PLUGIN -fplugin-arg-randomize_layout_plugin-performance-mode
> -Wframe-larger-than=2048 -fstack-protector -Wno-unused-but-set-variable
> -Wno-unused-const-variable -fno-var-tracking-assignments -pg -mfentry
> -DCC_USING_FENTRY -Wdeclaration-after-statement -Wno-pointer-sign
> -fno-strict-overflow -fno-merge-all-constants -fmerge-constants
> -fno-stack-check -fconserve-stack -Werror=implicit-int
> -Werror=strict-prototypes -Werror=date-time
> -Werror=incompatible-pointer-types -Werror=designated-init  -DMODULE 
> -DKBUILD_BASENAME='"bbswitch"'  -DKBUILD_MODNAME='"bbswitch"' -c -o
> /tmp/bbtest/bbswitch-0.8/.tmp_bbswitch.o /tmp/bbtest/bbswitch-0.8/bbswitch.c
> /tmp/bbtest/bbswitch-0.8/bbswitch.c: In function ‘acpi_call_dsm’:
> /tmp/bbtest/bbswitch-0.8/bbswitch.c:138:28: note: byref variable will be
> forcibly initialized
>          struct acpi_buffer buf = { ACPI_ALLOCATE_BUFFER, NULL };
>                             ^~~
> /tmp/bbtest/bbswitch-0.8/bbswitch.c:109:29: note: byref variable will be
> forcibly initialized
>      struct acpi_object_list input;
>                              ^~~~~
> /tmp/bbtest/bbswitch-0.8/bbswitch.c:108:24: note: byref variable will be
> forcibly initialized
>      struct acpi_buffer output = { ACPI_ALLOCATE_BUFFER, NULL };
>                         ^~~~~~
> /tmp/bbtest/bbswitch-0.8/bbswitch.c: In function ‘bbswitch_init’:
> /tmp/bbtest/bbswitch-0.8/bbswitch.c:398:28: note: byref variable will be
> forcibly initialized
>          struct acpi_buffer buf = { ACPI_ALLOCATE_BUFFER, NULL };
>                             ^~~
>    ./tools/objtool/objtool orc generate  --module --no-fp  --retpoline
> "/tmp/bbtest/bbswitch-0.8/.tmp_bbswitch.o";
>   if [ "-pg" = "-pg" ]; then if [ /tmp/bbtest/bbswitch-0.8/bbswitch.o !=
> "scripts/mod/empty.o" ]; then ./scripts/recordmcount 
> "/tmp/bbtest/bbswitch-0.8/bbswitch.o"; fi; fi;
> make -f ./scripts/Makefile.modpost
>   find /tmp/bbtest/bbswitch-0.8/.tmp_versions -name '*.mod' | xargs -r grep
> -h '\.ko$' | sort -u | sed 's/\.ko$/.o/' | scripts/mod/modpost -m -a -i
> ./Module.symvers -I /tmp/bbtest/bbswitch-0.8/Module.symvers  -o
> /tmp/bbtest/bbswitch-0.8/Module.symvers -S  -w  -s -T -
>   gcc -Wp,-MD,/tmp/bbtest/bbswitch-0.8/.bbswitch.mod.o.d  -nostdinc -isystem
> /usr/lib64/gcc/x86_64-mandriva-linux-gnu/7.3.0/include -I./arch/x86/include
> -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi
> -I./arch/x86/include/generated/uapi -I./include/uapi
> -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__
> -DCONFIG_CC_STACKPROTECTOR -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs
> -fno-strict-aliasing -fno-common -fshort-wchar
> -Werror-implicit-function-declaration -Wno-format-security -std=gnu11
> -fno-PIE -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64
> -falign-jumps=1 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387
> -mpreferred-stack-boundary=3 -mskip-rax-setup -mtune=generic -mno-red-zone
> -mcmodel=kernel -funit-at-a-time -DCONFIG_X86_X32_ABI -DCONFIG_AS_CFI=1
> -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -DCONFIG_AS_CFI_SECTIONS=1
> -DCONFIG_AS_FXSAVEQ=1 -DCONFIG_AS_SSSE3=1 -DCONFIG_AS_CRC32=1
> -DCONFIG_AS_AVX=1 -DCONFIG_AS_AVX2=1 -DCONFIG_AS_AVX512=1
> -DCONFIG_AS_SHA1_NI=1 -DCONFIG_AS_SHA256_NI=1 -pipe -Wno-sign-compare
> -fno-asynchronous-unwind-tables -mindirect-branch=thunk-extern
> -mindirect-branch-register -DRETPOLINE -fno-delete-null-pointer-checks
> -Wno-frame-address -Wno-format-truncation -Wno-format-overflow
> -Wno-int-in-bool-context -O2 --param=allow-store-data-races=0
> -DCC_HAVE_ASM_GOTO -fplugin=./scripts/gcc-plugins/latent_entropy_plugin.so
> -fplugin=./scripts/gcc-plugins/structleak_plugin.so
> -fplugin=./scripts/gcc-plugins/randomize_layout_plugin.so
> -DLATENT_ENTROPY_PLUGIN -fplugin-arg-structleak_plugin-verbose
> -fplugin-arg-structleak_plugin-byref-all -DSTRUCTLEAK_PLUGIN
> -DRANDSTRUCT_PLUGIN -fplugin-arg-randomize_layout_plugin-performance-mode
> -Wframe-larger-than=2048 -fstack-protector -Wno-unused-but-set-variable
> -Wno-unused-const-variable -fno-var-tracking-assignments -pg -mfentry
> -DCC_USING_FENTRY -Wdeclaration-after-statement -Wno-pointer-sign
> -fno-strict-overflow -fno-merge-all-constants -fmerge-constants
> -fno-stack-check -fconserve-stack -Werror=implicit-int
> -Werror=strict-prototypes -Werror=date-time
> -Werror=incompatible-pointer-types -Werror=designated-init 
> -DKBUILD_BASENAME='"bbswitch.mod"'  -DKBUILD_MODNAME='"bbswitch"' -DMODULE 
> -c -o /tmp/bbtest/bbswitch-0.8/bbswitch.mod.o
> /tmp/bbtest/bbswitch-0.8/bbswitch.mod.c
>   ld -r -m elf_x86_64 -z max-page-size=0x200000 -T
> ./scripts/module-common.lds --build-id  -o
> /tmp/bbtest/bbswitch-0.8/bbswitch.ko /tmp/bbtest/bbswitch-0.8/bbswitch.o
> /tmp/bbtest/bbswitch-0.8/bbswitch.mod.o ;  true
> ld: warning: Disabling --icf due to -r
> make[1]: Leaving directory '/usr/src/linux-4.16.13-desktop-1omv'

So it build with some warnings but no errors.

Now if you do a ls in that folder you should 
have the .ko module build, right ?

So if you want to give it an test now run: 

sudo insmod ./bbswitch.ko

If that works , then something is wrong with dkms.

## Comment 18263
Date: 2018-06-09 16:42:08 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Yes. There is a bbswitch.ko and insmod goes fine.

## Comment 18264
Date: 2018-06-09 16:45:17 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

But bbswitch.ko is not in the kernel directory,

/lib/modules/4.16.13-desktop-1omv/kernel/drivers/acpi/

## Comment 18265
Date: 2018-06-09 17:12:18 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to adelson from comment #19)
> But bbswitch.ko is not in the kernel directory,
> 
> /lib/modules/4.16.13-desktop-1omv/kernel/drivers/acpi/

Yes it can't be :) You used it directly from the build dir.

Thx for testing that .. It shows dkms is doing some stupid things.

## Comment 18266
Date: 2018-06-09 19:05:09 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Should I file a bug on a dkms-bbswitch problem? I wonder if it wouldn't be better to report at bbswitch git-Hub site.

## Comment 18278
Date: 2018-06-09 22:43:35 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to adelson from comment #21)
> Should I file a bug on a dkms-bbswitch problem? I wonder if it wouldn't be
> better to report at bbswitch git-Hub site.

No and no :)

I think the bug likes in dkms itself somewho or packaging.

## Comment 18281
Date: 2018-06-10 02:30:30 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

Well it used to work with older kernels ...

## Comment 18299
Date: 2018-06-16 11:09:53 +0000
From: Marco &lt;<m-tomatis@tiscali.it>&gt;

Created attachment 1613
build log file

> Attached file: make.log (text/plain, 845285 bytes)
> Description:   build log file

## Comment 18300
Date: 2018-06-16 11:18:50 +0000
From: Marco &lt;<m-tomatis@tiscali.it>&gt;

I'm experiencing the same dkms issue with nvidia340 module: Error!  Build of nvidia.ko.gz failed for: 4.16.13-desktop-1omv (x86_64)
The file nvidia-uvm.ko can be created, but the dkms won't install the nvidia module under this new kernel anyway. Maybe it is a kernel issue.

## Comment 18301
Date: 2018-06-16 19:38:51 +0000
From: blackcrack &lt;<blackcrack@blackysgate.de>&gt;

i guess the script/patch for the nvidia must be updated like 
Colin it have every done, because the patch was belong 
the older kernel, now it's there a more new kernel 4.16 
and no more 4.15 for a good compiling..  the patch should 
changed and adust belong the kernel and nvidia-version 
like patch-nv-kernel-4.15-1.0.0.omv.rpm and patch-nv-kernel-4.16-1.0.0.omv.rpm
and so on .. if can make in this direction ..

i wait also on a "update" or fresh patch for the kernel versions 4.16

because the error /issue is show on my computers also..
indeed, confirmed!

best regards
blacky

## Comment 18302
Date: 2018-06-19 20:57:52 +0000
From: @benbullard79

(In reply to Marco from comment #25)
> I'm experiencing the same dkms issue with nvidia340 module: Error!  Build of
> nvidia.ko.gz failed for: 4.16.13-desktop-1omv (x86_64)
> The file nvidia-uvm.ko can be created, but the dkms won't install the nvidia
> module under this new kernel anyway. Maybe it is a kernel issue.

NO, for the 10,000th time this is not a kernel issue. It is an issue with nvida software that has existed since the beginning. And this issue in OpenMandriva has been well covered in our forum and our blog. From: 

https://www.openmandriva.org/en/news/article/why-nvidia-drivers-sometimes-don-t-build-when-new-kernel-is-introduced

"nvidia driver releases are designed to work up to a specific kernel version and often need to be patched to work with later kernel versions. Be aware that nVidia is closed source so these patches are mostly provided by nVidia or other kernel hackers. Whatever the source of the patches they are not usually available until after a new kernel is released. It is therefore inevitable that in some instances there will be a delay between the time when a new kernel is released by OpenMandriva and when a new nVidia driver package can be made available."

To say this in other words this in not a kernel issue. It is an issue that nvidia software has had since forever. This issue will exist as long as nVidia the company does not correct it. Please complain to nVidia the company.

## Comment 18303
Date: 2018-06-19 21:17:17 +0000
From: @benbullard79

This issue is IN_PROGRESS. 

https://forum.openmandriva.org/t/nvidia-drivers/1890

The issue is the same as it has always been with nVidia drivers. They are written to work up to a certain kernel version. When a new kernel is released the older nvidia driver packages will not work. It is that simple and it has always been this way. 

So when nVidia users see a new kernel version and DON'T see new nvida software why do they keep trying to build something that everyone in Linux World knows will not work? When there is a new Linux kernel there has to be new nVida software. Every-time. Until that software is built in your Linux distribution your choices as an nVidia user are two.

1. Use the previous kernel version that works with nVidia proprietary packages.

OR:

2. Use the newer kernel with open-source (nouveau) driver.

The issue of how long it takes OpenMandriva to build new nVidia packages is a separate issue. It will not change until users bother to talk directly to developers about this. I do know that Colin tries very hard to keep these up to date. But Colin is not always available to do this, plus he can't do this until nVidia the company makes the new software available, plus sometimes there are issues building new packages and it takes more time.

The above Forum thread is the perfect opportunity to talk directly to developers about this. Make some noise users.

## Comment 18304
Date: 2018-06-20 06:30:37 +0000
From: blackcrack &lt;<blackcrack@blackysgate.de>&gt;

you say really to the peoples in the open Statement, 
they have no choice and they have to take this what 
you (or other peoples in this distro) want/the distro 
and not the users ? 
And you want the peoples say, what they have to buy ?

wow...!

2nd, Bero does at all and we have no other developer.. 
(reasons why we have no more developers ?)

i can only say, a dog is not a penguin.. ..

best regards
Blacky

## Comment 18305
Date: 2018-06-20 10:57:22 +0000
From: Marco &lt;<m-tomatis@tiscali.it>&gt;

Ok, the nvidia driver is proprietary code. But the legacy drivers have been the same for years, so it is absurd they should be updated by Nvidia every time a new kernel comes out. It is up to the kernel developers understand the reasons for a module malfunction and find a suitable workaround.

## Comment 18306
Date: 2018-06-20 11:20:35 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to Marco from comment #30)
> Ok, the nvidia driver is proprietary code. But the legacy drivers have been
> the same for years, so it is absurd they should be updated by Nvidia every
> time a new kernel comes out. It is up to the kernel developers understand
> the reasons for a module malfunction and find a suitable workaround.

No you are wrong .. 'kernel developers' only cares about in-tree drivers.

It was , it is and will be only like this.

While one could fix/workaround some code to 'compile' , the one who
fixes that has not a clue whatever the driver still works with his changes,
since the other part is closed source and so no way to verify it breaks something.

Closed source means : 'The one who owns the code have to fix..'

## Comment 18307
Date: 2018-06-20 11:26:20 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to blackcrack from comment #29)
> you say really to the peoples in the open Statement, 
> they have no choice and they have to take this what 
> you (or other peoples in this distro) want/the distro 
> and not the users ? 
> And you want the peoples say, what they have to buy ?
> 

Hmm ? That is known forever.. Nvidia HW on Linux is bad choice.

And again you seems to not get it.. Distros takes whatever
Nvidia releases .. and if is broken not much to do.

You should complain to Nvidia itself , right ? :P

## Comment 18308
Date: 2018-06-20 13:13:39 +0000
From: @benbullard79

(In reply to Marco from comment #30)
> Ok, the nvidia driver is proprietary code. But the legacy drivers have been
> the same for years, so it is absurd they should be updated by Nvidia every
> time a new kernel comes out. It is up to the kernel developers understand
> the reasons for a module malfunction and find a suitable workaround.

You've hit on one problem. OpenMandriva does not really have a kernel developer.

## Comment 18309
Date: 2018-06-20 13:20:14 +0000
From: @benbullard79

(In reply to blackcrack from comment #29)
> you say really to the peoples in the open Statement, 
> they have no choice and they have to take this what 
> you (or other peoples in this distro) want/the distro 
> and not the users ? 

> 
> best regards
> Blacky

No I'm repeating to users what nVidia says to you. It is and always has been nVidia that decides this. Only nVidia can change this.

## Comment 18310
Date: 2018-06-20 14:31:47 +0000
From: @benbullard79

Let me clarify/expand my comments in Comment # 33. OpenMandriva does not have a kernel specialist or a graphic hardware specialist. 

So the original problem is with nVidia the company. But the reason it is more obvious to users in OM vs. Mandriva is the difference in staff. Mandriva had a staff that included full time employees, developers, and bosses and lots, and lots of part time developers. I don't know for sure what the number of devs was but suffice to say probably 10 times or greater than what OM has. OM has a group of volunteers that are all part timers and no official staff or bosses. No paid developers. There is no one in OM presently than can "make" a developer do anything. So obviously Mandriva could keep problems like this out of view of most users much more easily than OM can. With 6-8 part timers sometimes it is taking weeks after a kernel is released before the nVidia packages and sometimes other out-of-tree kernel module packages are built.

I have proposed a Packaging Rule about this:

https://forum.openmandriva.org/t/are-nvidia-drivers-the-squeaky-wheel/1894/3

but this was only published yesterday so no response yet on that. Even with such a rule users can expect this process to take longer in OM than it did with Mandriva. Community distro... Either people in the Community step up and do things or things don't get done.

## Comment 18311
Date: 2018-06-21 08:46:33 +0000
From: blackcrack &lt;<blackcrack@blackysgate.de>&gt;

well and why do we not use the way what's works,
use an 1st Penguin, 2nd let the peoples come and help belong the mandrake/Mandriva way and make a rolling release (what it is in planing now finally)

a baiting dog helps not to stay on the way of the Penguin and the Penguin was the way who have bring all together. 

                           .........
                           Signation
                           .........
          ---===<<<((([[[Blackcrack]]])))>>>===---

## Comment 18312
Date: 2018-06-21 08:48:50 +0000
From: blackcrack &lt;<blackcrack@blackysgate.de>&gt;

and update/maintain the Draktools include the own and orginal Installer !
(did i almost forgotten) the way in the last years was for the garbage.

## Comment 18316
Date: 2018-06-22 09:40:48 +0000
From: Marco &lt;<m-tomatis@tiscali.it>&gt;

I have understood that Nvidia drivers must be taken as they are. But the kernel has changed in a way that what worked before now does not compile anymore.
So, is someone able to understand the reason for this even without a kernel developer? What has changed in the 4.16.13 kernel so much to prevent nvidia modules from building?

## Comment 18317
Date: 2018-06-22 12:11:46 +0000
From: blackcrack &lt;<blackcrack@blackysgate.de>&gt;

Hi Marco,

the versionsnumber of the kernel "4.15" to "4.16" which change also the folders and other things, i guess this is the reason ..
there was every a patch for it to be in able for clean compiling, 
this must be update(ing), so the patch, like i guess..
best regards
Blacky

## Comment 18318
Date: 2018-06-22 14:00:47 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to Marco from comment #38)
> I have understood that Nvidia drivers must be taken as they are. But the
> kernel has changed in a way that what worked before now does not compile
> anymore.
> So, is someone able to understand the reason for this even without a kernel
> developer? What has changed in the 4.16.13 kernel so much to prevent nvidia
> modules from building?

Yes is simple .. kernel 4.16 and newer changed the way 
DMA OPS and SWIOLTB works. While Nvidia still uses code got 
changed or removed it does not work.

Latest version of Nvidia release works again as is with kernel 4.16 
and newer.. 

I think Colin is working on an update of the Nvidia package already.

## Comment 18319
Date: 2018-06-22 19:58:13 +0000
From: Marco &lt;<m-tomatis@tiscali.it>&gt;

Thank you all for your explanation. It something like a code misalignment between new kernel and old drivers.

## Comment 18447
Date: 2018-07-27 02:56:19 +0000
From: adelson &lt;<adelson.oliveira@gmail.com>&gt;

There has been issued a new nvidia driver 390.77 which is supposed to support newer linux kernels (up to 4.17)

The build of such a nvidia driver would fix related problems and this very issue.

Thus, I would like to suggest changing this bug to an enhancement one where it is suggested the building of such nvidia 390.77 driver for OMV LX 3.0x.

Thanks

## Comment 18448
Date: 2018-07-27 02:59:44 +0000
From: @benbullard79

(In reply to adelson from comment #42)
> There has been issued a new nvidia driver 390.77 which is supposed to
> support newer linux kernels (up to 4.17)
> 
> The build of such a nvidia driver would fix related problems and this very
> issue.
> 
> Thus, I would like to suggest changing this bug to an enhancement one where
> it is suggested the building of such nvidia 390.77 driver for OMV LX 3.0x.
> 
> Thanks

Let's do a new but report instead and leave this one as is for a bit. I'll do it myself.

## Comment 18449
Date: 2018-07-27 03:03:06 +0000
From: @benbullard79

(In reply to adelson from comment #42)
> There has been issued a new nvidia driver 390.77 which is supposed to
> support newer linux kernels (up to 4.17)
> 
> The build of such a nvidia driver would fix related problems and this very
> issue.
> 
> Thus, I would like to suggest changing this bug to an enhancement one where
> it is suggested the building of such nvidia 390.77 driver for OMV LX 3.0x.
> 
> Thanks

I'll do a new bug report for that. And I'm marking as RESOLVED/CANTFIX since the 390.25 driver was never built for kernel 4.16.13. Or at least that is what I understand.

## Comment 18450
Date: 2018-07-27 03:03:18 +0000
From: @benbullard79

(In reply to adelson from comment #42)
> There has been issued a new nvidia driver 390.77 which is supposed to
> support newer linux kernels (up to 4.17)
> 
> The build of such a nvidia driver would fix related problems and this very
> issue.
> 
> Thus, I would like to suggest changing this bug to an enhancement one where
> it is suggested the building of such nvidia 390.77 driver for OMV LX 3.0x.
> 
> Thanks

I'll do a new bug report for that. And I'm marking as RESOLVED/CANTFIX since the 390.25 driver was never built for kernel 4.16.13. Or at least that is what I understand.
